examples/common/mcstack/FairStack.cxx

```diff
@@ -239,8 +239,7 @@ void FairStack::FillTrackArray()
 
 Int_t FairStack::GetCurrentTrackNumber() const
 {
-    std::map<Int_t, Int_t>::const_iterator tempIter =
-        fFSTrackMap.find(fCurrentTrack);   // check if the track was created by FastSimulation
+    auto tempIter = fFSTrackMap.find(fCurrentTrack);   // check if the track was created by FastSimulation
     if (tempIter != fFSTrackMap.end()) {
         return tempIter->second;
     }
```

Comment:
```suggestion
    // check if the track was created by FastSimulation
    if (auto it = fFSTrackMap.find(fCurrentTrack); it != fFSTrackMap.end()) {
        return it->second; 
    }
```

Limit the variable scope even further? see Examples @ https://en.cppreference.com/w/cpp/container/map/find

Reply:
In general, I like limiting variable scopes!

Just we're back to the old problem of the convoluted if statement. I'd like the assignment and the check to be on different lines inside the `if`. And last time we checked, this involved turning of the clang-format. Not exactly what I like.

If the above if-statement-style is wanted, I am of course happy to change it. :-)

Reply:
as discussed on mm: https://clang.llvm.org/docs/ClangFormatStyleOptions.html#bracewrapping with `BWACS_MultiLine` should do the trick?

Reply:
As #1586 was merged, I have rebased this PR and updated most places looking like this one.